pandablocks/commands.py

```diff
@@ -229,49 +239,123 @@ def execute(self) -> ExchangeGenerator[None]:
         assert ex.line == "OK"
 
 
-class GetBlockNumbers(Command[Dict[str, int]]):
-    """Get the name and number of each block type in a dictionary,
-    alphabetically ordered
+@dataclass
+class GetBlockInfo(Command[Dict[str, BlockInfo]]):
+    """Get the name, number, and description of each block type
+    in a dictionary, alphabetically ordered
+
+    Args:
+        skip_description: If `True`, prevents retrieving the description
+            for each Block. This will reduce network calls.
 
     For example::
 
-        GetBlockNumbers() -> {"LUT": 8, "PCAP": 1, ...}
+         GetBlockInfo() ->
+             {
+                 "LUT": BlockInfo(number=8, description="Lookup table"),
+                 "PCAP": BlockInfo(number=1, description="Position capture control"),
+                 ...
+             }
     """
 
-    def execute(self) -> ExchangeGenerator[Dict[str, int]]:
+    skip_description: bool = False
+
+    def execute(self) -> ExchangeGenerator[Dict[str, BlockInfo]]:
         ex = Exchange("*BLOCKS?")
         yield ex
-        blocks_list = []
+
+        blocks_list, commands = [], []
         for line in ex.multiline:
             block, num = line.split()
             blocks_list.append((block, int(num)))
-        return dict(sorted(blocks_list))
+            commands.append(Get(f"*DESC.{block}"))
+
+        if self.skip_description:
+            description_values = tuple(None for _ in range(len(commands)))
+        else:
+            description_values = yield from _execute_commands(*commands)
+
+        blocks_info = {
+            block[0]: BlockInfo(number=block[1], description=desc)
+            for block, desc in zip(blocks_list, description_values)
+        }
+
+        return OrderedDict(sorted(blocks_info.items()))
```

Comment:
```suggestion
        block_infos = {
            block: BlockInfo(number=num, description=desc)
            for (block, num), desc in sorted(zip(blocks_list, description_values))
        }

        return block_infos
```
I think we should change this to a dict (which is guaranteed to keep insertion ordering since python3.7). We depend on this in other places in the code. We can still use the OrderedDict in tests to compare against it:
```
>>> OrderedDict({1: 2, 3: 4}) == OrderedDict({3: 4, 1: 2})
False
```